### PR TITLE
Fixed issue: for non-admin user global setting left panel should visible properly  #6710

### LINKED ISSUE #6710: for non-admin user global setting left panel should visible properly
global setting left panel for non-admin user need to show option which is accessible to normal user

image

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on the non-admin user global setting left panel should be visible properly 
Fixes #6710

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/AddRolePage/AddRolePage.tsx [hunk 1]
```diff
@@ -18,21 +18,15 @@ import { useHistory } from 'react-router-dom';
...
 import { showErrorToast } from '../../../utils/ToastUtils';
 const { Option } = Select;
 const breadcrumb = [
   {
     name: 'Roles',
-    url: getSettingPath(
-      GlobalSettingsMenuCategory.ACCESS,
-      GlobalSettingOptions.ROLES
-    ),
+    url: getPath(GlobalSettingOptions.ROLES),
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/AddRolePage/AddRolePage.tsx:29]: Can we take this to constant and use it here and in other places?
```js
const rolesPath=getPath(GlobalSettingOptions.ROLES)
```
```suggestion
    url: rolesPath,
```

File: openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/AddRolePage/AddRolePage.tsx [hunk 2]
```diff
@@ -58,12 +52,7 @@ const AddRolePage = () => {
   };
 
   const handleCancel = () => {
-    history.push(
-      getSettingPath(
-        GlobalSettingsMenuCategory.ACCESS,
-        GlobalSettingOptions.ROLES
-      )
-    );
+    history.push(getPath(GlobalSettingOptions.ROLES));
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/AddRolePage/AddRolePage.tsx:55]:
```suggestion
    history.push(rolesPath);
```

File: openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx [hunk 3]
```diff
@@ -40,17 +40,15 @@ export const getGlobalSettingMenus = (
     icon,
     children: children
       ? children
-          .filter(({ isProtected }) => {
-            return hasAccess ?? !isProtected;
-          })
-          .map(({ label, icon }) =>
-            getGlobalSettingMenus(
+          .filter((menu) => (hasAccess ? menu : !menu.isProtected))
+          .map(({ label, icon }) => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx:44]: If possible please avoid negative conditions.

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx:44]: I don't think so we can avoid it in this case as we need to send a false for filtering out

**Pull request merged**